### ISSUE TITLE
[Bug] Wrong value for perceiver_cond_length_compression (256 instead of 1024)

### ISSUE THREAD

**Issue by talipturkmen** [coqui-ai/TTS#3631]
### Describe the bug

[Bug] 

Hello,

In XTTSv2, in dataloader condition indices (condition start indice and end indice) are loading as audio samples which later will be compressed by 256 due to mel spectrogram extraction. Later these condition start and end indices are used to mask the ground truth audio codes. Since, audio codes compressed by 1024 times these start and end indices also has to be divided by 1024 but they are divided by perceiver_cond_length_compression  which is set to 256 coming from the default args of GPT. In this case condition indices in dvae domain will refer to 4 times higher of what it should refer. So you will mask the wrong part of the target audio tokens. I couldn't find any place where they are set to 1024 correcty and I can not understand with this bug how it's able to train and finetune. 

I'd appreciate if anyone shed lights on this topic.

Error lines:
https://github.com/coqui-ai/TTS/blob/dbf1a08a0d4e47fdad6172e433eeb34bc6b13b4e/TTS/tts/layers/xtts/gpt.py#L110
https://github.com/coqui-ai/TTS/blob/dbf1a08a0d4e47fdad6172e433eeb34bc6b13b4e/TTS/tts/layers/xtts/gpt.py#L414

### To Reproduce

- xttsv2 finetuning

### Expected behavior

...

### Logs

```shell
...
```

### Environment

```shell
- all xttsV2 versions
```

**Comment by Edresson** [comment]
Hi @talipturkmen, 

The conditioning input reducing factor when using the Perceiver Resampler is just 256 because we extract mel spectrogram with  hop size equal to 256, check here: https://github.com/coqui-ai/TTS/blob/dbf1a08a0d4e47fdad6172e433eeb34bc6b13b4e/TTS/tts/layers/xtts/trainer/gpt_trainer.py#L147

Best regards,
Edresson Casanova

**Comment by talipturkmen** [comment]
Thank you for the explanation. Now I get it.
